bee-block/src/output/unlock_condition/mod.rs

```diff
@@ -240,10 +240,23 @@ impl UnlockConditions {
         address: &'a Address,
         milestone_index: MilestoneIndex,
         milestone_timestamp: u32,
-    ) -> &'a Address {
-        self.expiration()
-            .and_then(|e| e.return_address_expired(milestone_index, milestone_timestamp))
-            .unwrap_or(address)
+    ) -> Option<&'a Address> {
+        match self.expiration() {
+            Some(expiration) => expiration.return_address_expired(milestone_index, milestone_timestamp),
```

Comment:
Is this correct ? If the unlock is not yet expired shouldn't it return the unlock address ?

Reply:
Ah right, thinking about how to do this best, since there can also be the case where no address can be returned

Reply:
Yeah I guess `return_address_expired` itself needs to be changed 🤔